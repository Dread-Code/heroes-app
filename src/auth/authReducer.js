import { LOGIN, LOGOUT } from "../types/types";

export const authReducer = (state = {}, action) => {
  switch (action?.type) {
    case LOGIN:
      return {
        ...action.payload,
        logged: true,
      };
    case LOGOUT:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
