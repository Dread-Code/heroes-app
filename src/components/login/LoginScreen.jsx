import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { LOGIN } from "../../types/types";

export const LoginScreen = () => {
  const { dispatch } = useContext(AuthContext);

  /**
   * History es un propiedad que nos ayuda
   * a accesader a ciertos metodos de nuestro navegador
   * o de react.router
   */
  const handleLOgin = () => {
    // push nos redrecciona a una de las rutas
    // history.push("/");
    // replace remplaza el layout y no permitira volver
    // a atras al mismo layout
    // history.replace("/");
    dispatch({
      type: LOGIN,
      payload: {
        name: "Lucas",
        email: "lucas@gmail.com",
        id: 23,
      },
    });
  };

  return (
    <div className="container mt-5">
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLOgin}>
        login
      </button>
    </div>
  );
};
