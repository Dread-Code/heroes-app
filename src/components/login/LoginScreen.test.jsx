import React from "react";
import { mount } from "enzyme";
import { LoginScreen } from "./LoginScreen";
import { AuthContext } from "../../auth/AuthContext";

describe("LoginScreen", () => {
  const contextValue = {
    dispatch: jest.fn(),
  };

  const historyMock = {
    replace: jest.fn(),
  };
  test("should match snapshoot", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen history={historyMock} />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("should dispatch and make the replace", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen history={historyMock} />
      </AuthContext.Provider>
    );
    const handleClick = wrapper.find("button").prop("onClick");

    handleClick();
    expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
    expect(historyMock.replace).toHaveBeenCalledTimes(1);

    // EMULAMOS EL local storage
    localStorage.setItem("lastpath", "/dc");
    handleClick();
    expect(historyMock.replace).toHaveBeenCalledWith("/dc");
  });
});
