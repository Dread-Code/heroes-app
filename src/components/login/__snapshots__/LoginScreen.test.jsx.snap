// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should match snapshoot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
